crates/codegen/language/definition/src/model/types.rs

```diff
@@ -99,18 +117,33 @@ mod wrapper {
         pub allow_empty: Option<bool>,
     }
 
-    #[derive(Debug, Eq, PartialEq, Serialize)]
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize)]
     pub struct SeparatedItem {
         pub name: Identifier,
-        pub separated: Identifier,
+        pub separated: Separated,
```

Comment:
IIUC, this is done to allow empty tuple deconstruction. But for scope graphs, we always need to create a tuple node anyways, even if it is empty. I suggest reverting that change, keeping the original `TupleValue`, as we will attach stuff to it.

Additionally, it would be much easier for CST->AST conversion if we always expect `Separated` children to be in a specific order, without having to guess which is which.

Reply:
Reverted this - I'll see if I can fix the test regression in the CST output, instead.